chia/cmds/plotnft_funcs.py

```diff
@@ -9,47 +10,68 @@
 from chia.rpc.farmer_rpc_client import FarmerRpcClient
 from chia.rpc.wallet_rpc_client import WalletRpcClient
 from chia.types.blockchain_format.sized_bytes import bytes32
-from chia.util.bech32m import encode_puzzle_hash
+from chia.util.bech32m import encode_puzzle_hash, decode_puzzle_hash
 from chia.util.byte_types import hexstr_to_bytes
 from chia.util.config import load_config
 from chia.util.default_root import DEFAULT_ROOT_PATH
-from chia.util.ints import uint16
+from chia.util.ints import uint16, uint32
 from chia.wallet.transaction_record import TransactionRecord
 from chia.wallet.util.wallet_types import WalletType
-import aiohttp
 
 
-async def create(args: dict, wallet_client: WalletRpcClient, fingerprint: int) -> None:
-    pool_url = args["pool_url"]
+async def create_pool_args(pool_url: str) -> Dict:
     try:
         async with aiohttp.ClientSession() as session:
             async with session.get(f"{pool_url}/pool_info") as response:
                 if response.ok:
                     json_dict = json.loads(await response.text())
                 else:
-                    print(f"Response not OK: {response.status}")
-                    return
+                    raise ValueError(f"Response from {pool_url} not OK: {response.status}")
     except Exception as e:
-        print(f"Error connecting to pool {pool_url}: {e}")
-        return
+        raise ValueError(f"Error connecting to pool {pool_url}: {e}")
 
     if json_dict["relative_lock_height"] > 1000:
-        print("Relative lock height too high for this pool, cannot join")
-        return
+        raise ValueError("Relative lock height too high for this pool, cannot join")
     if json_dict["protocol_version"] != POOL_PROTOCOL_VERSION:
-        print(f"Incorrect version: {json_dict['protocol_version']}, should be {POOL_PROTOCOL_VERSION}")
-        return
+        raise ValueError(f"Incorrect version: {json_dict['protocol_version']}, should be {POOL_PROTOCOL_VERSION}")
 
-    print(f"Will create a plot NFT and join pool: {pool_url}.")
+    header_msg = f"\n---- Pool parameters fetched from {pool_url} ----"
+    print(header_msg)
     pprint(json_dict)
+    print("-" * len(header_msg))
+    return json_dict
+
+
+async def create(args: dict, wallet_client: WalletRpcClient, fingerprint: int) -> None:
+    state = args["state"]
+
+    # Could use initial_pool_state_from_dict to simplify
+    if state == "SELF_POOLING":
+        pool_url: Optional[str] = None
+        relative_lock_height = uint32(0)
+        new_address = await wallet_client.get_next_address("1", True)
```

Comment:
Can we just not provide an address and let the wallet backend do it? That means less requests from the CLI.

Reply:
Yes, that's what I wanted to do too - the only reason I did it this way is that I was wondering about a use case like this: you want to create and control a plotnft, but you want it to pay out to a friend for some reason. (Of course, you can always redirect it later).

I admit that's a bit of a stretch, and it's easy to add later, so I think I'll change it to what you are suggesting.